comp/process/bundle_test.go

```diff
@@ -47,13 +47,11 @@ func TestBundleDependencies(t *testing.T) {
 		fx.Supply(mockCoreBundleParams),
 		fx.Provide(func() types.CheckComponent { return nil }),
 		core.MockBundle(),
-		workloadmetafx.Module(workloadmeta.NewParams()),
+		workloadmetafxmock.MockModule(workloadmeta.NewParams()),
+		fx.Provide(func() tagger.Component { return taggerfxmock.SetupFakeTagger(t) }),
```

Comment:
This and the case below look a bit strange. `SetupFakeTagger` takes care of injecting workloadmeta and other dependencies so the line above shouldn't be necessary.

Given that the dependencies are already defined here maybe we could use `taggerfxmock.MockModule()` instead.

Reply:
Given the error:
```
missing type: - workloadmeta.Component (did you mean to Provide it?)
```

Is this comment addressed by calling `fx.Provide` here? https://github.com/DataDog/datadog-agent/blob/114fdf32beeb00e453666a9f961b003e9ee6ebe2/comp/core/tagger/fx-mock/fx.go#L38

Reply:
No, as this collides with `SetupFakeTagger`, which also needs to have it.
```
        Failed: cannot provide function "reflect".makeFuncStub (/usr/local/go/src/reflect/asm_arm64.s:29): cannot provide mock.Mock from [0].Mock: already provided by "reflect".makeFuncStub (/usr/local/go/src/reflect/asm_arm64.s:29)
```

Reply:
Those FX issues is something that we will address in future PRs 👍